src/cmd-buildextend-extensions-container

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3 -u
+
+import argparse
+import os
+import shutil
+import sys
+
+import createrepo_c as cr
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
+
+from cosalib import cmdlib
+from cosalib.builds import Builds
+from cosalib.meta import GenericBuildMeta
+
+
+def main():
+    args = parse_args()
+
+    workdir = os.path.abspath(os.getcwd())
+
+    builds = Builds()
+    builddir = builds.get_build_dir(args.build)
+    buildmeta = GenericBuildMeta(workdir=workdir, build=args.build)
+
+    if 'extensions-container' in buildmeta and not args.force:
+        print(f"Extensions container already exist: {buildmeta['extensions-container']['path']}")
+        print("Use --force to force a rebuild")
+        return
+
+    # For rhcos we expect we copy the extentions folder along with 
+    # extensions.yaml and manifest.yaml to the config dir.
+    context_dir = 'src/config/'
+    treefile_src = context_dir + 'manifest.yaml'
+    extensions_src = context_dir + 'extensions.yaml'
+    extensions_dockerfile = context_dir + 'extensions/Dockerfile'
+    if not os.path.exists(extensions_dockerfile):
+        raise Exception(f"Missing {extensions_dockerfile}")
+    if not os.path.exists(extensions_src):
+        raise Exception(f"Missing {extensions_src}")
+
+    ## Call buildah/podman inside a vm with src/config as context.
+    container_built = run_container_build(context_dir)
+    if not container_built:
+        return
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--build", help="Build ID", default='latest')
+    parser.add_argument("--force", help="Force rebuild", action='store_true')
+    return parser.parse_args()
+
+def run_container_build(context_dir):
+    print("call runvm and podman inside it...")
```

Comment:
Yeah, all this logic is in shell (`cmdlib.sh`) today unfortunately :cry: 

But, combining this and my comment about https://github.com/coreos/coreos-assembler/pull/2919 - today the cosa Go library has good support for running code from `cmdlib.sh`.